src/Kestrel/KestrelServerOptionsConfigurationExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core
+{
+    public static class KestrelServerOptionsConfigurationExtensions
+    {
+        public static KestrelConfigBuilder Configure(this KestrelServerOptions options, IConfiguration config)
+        {
+            return new KestrelConfigBuilder(options, config); // Assigns itself to options.ConfigurationBuilder
```

Comment:
Why not just create a local variable and assign it to options.ConfigurationBuilder in this method instead? It feels slightly better than assigning a partially constructed object even if it's functionally the same.

Reply:
I did it this way because Build then un-assigns itself, this way you can see the whole lifecycle in one class. I can change it.